core/src/main/scala/org/apache/spark/status/api/v1/StagesResource.scala

```diff
@@ -204,7 +212,7 @@ private[v1] class StagesResource extends BaseAppResource {
       pageLength = queryParameters.getFirst("length").toInt
     }
     withUI(_.store.taskList(stageId, stageAttemptId, pageStartIndex, pageLength,
-      indexName(columnNameToSort), isAscendingStr.equalsIgnoreCase("asc")))
+      indexName(columnNameToSort), "asc".equalsIgnoreCase(isAscendingStr)))
```

Comment:
`isAscendingStr` may be NPE if URI was encoded twice.
After change here, at least the stage page can be opened now